probatus/missing/imputation.py

```diff
@@ -0,0 +1,318 @@
+# Copyright (c) 2021 ING Bank N.V.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+# the Software, and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+from probatus.utils import preprocess_data, preprocess_labels,BaseFitComputePlotClass,get_single_scorer
+from  sklearn.model_selection import cross_val_score
+from  sklearn.pipeline import Pipeline
+from  sklearn.impute import SimpleImputer
+from  sklearn.compose import ColumnTransformer
+from  sklearn.preprocessing import OneHotEncoder
+import matplotlib.pyplot as plt
+import numpy as np 
+import pandas as pd 
+
+class CompareImputationStrategies(BaseFitComputePlotClass):
+    """
+    Comparison of various imputation stragegies that can be used for imputation 
+    of missing values. 
+    The aim of this class is to present the model performance based on imputation
+    strategies and choosen model.
+    For models like XGBoost & LighGBM which have capabilities to handle misisng values by default
+    the model performance with no imputation will be shown as well.
+    The missing values categorical features are filled with `missing` and an missing indicator is
+    added.
+
+    Usage E.g.
+    ```python
+
+    #Import the class
```

Comment:
In the example you have to also add imports and data creation. We have unit tests that check if the example codes run.

You should add this class to the `CLASSES_TO_TEST` in https://github.com/ing-bank/probatus/blob/main/tests/docs/test_docstring.py

Same applies to the notebook in:
https://github.com/ing-bank/probatus/blob/main/tests/docs/test_notebooks.py

Reply:
Done. Did not know about this, but this is a cool feature.